podjs/src/browserPod.ts

```diff
@@ -159,30 +159,41 @@ class LocalStoragePolyOut implements PolyOut {
         });
     }
 
-    readdir(path: string): Promise<string[]> {
+    readdir(id: string): Promise<{ [key: string]: string }[]> {
         files = new Map<string, Stats>(
             JSON.parse(localStorage.getItem(BrowserPolyNav.filesKey) || "[]")
         );
+        console.log({ files });
         return new Promise((resolve, reject) => {
             const filteredFiles = Array.from(files)
-                .filter((file) => file[0].startsWith(path))
+                .filter((file) => file[0].startsWith(id))
                 .map((file) => file[0]);
-
-            if (path == "") {
-                resolve(filteredFiles);
+            const retList: { [key: string]: string }[] = [];
+            if (id == "") {
+                for (let i = 0; i < filteredFiles.length; i++) {
+                    const file = filteredFiles[i];
+                    retList.push({ id: file, path: "" });
+                }
+                resolve(retList);
```

Comment:
Similarly to the Kotlin code, this could be simplified:

```
if (id == "") {
    resolve(filteredFiles.map((file) => ({ id: file, path: file })));
}
```

(And yes, I don't think `path` should be `""`, by my understanding ID and relative path are just identical in this case.)